beacon/engine/types.go

```diff
@@ -297,10 +297,16 @@ func ExecutableDataToBlockNoHash(data ExecutableData, versionedHashes []common.H
 		withdrawalsRoot = &h
 	}
 
+	isthmusEnabled := bType.IsIsthmus(data.Timestamp)
 	var requestsHash *common.Hash
 	if requests != nil {
+		if isthmusEnabled && len(requests) > 0 {
+			return nil, fmt.Errorf("requests should be empty for Isthmus blocks")
+		}
 		h := types.CalcRequestsHash(requests)
 		requestsHash = &h
+	} else if isthmusEnabled {
+		return nil, fmt.Errorf("requests should be non-nil for Isthmus blocks")
```

Comment:
Can we word this more closely to [the specs](https://github.com/ethereum-optimism/specs/blob/509b58bf8b68049f292ea19686ac9647340a5925/specs/protocol/exec-engine.md#engine_newpayloadv4) saying: "requests must be an empty array for Isthmus blocks"